utils/db/users.ts

```diff
@@ -158,4 +158,34 @@ export const updateUser = async (userId: string, user: DbUser) => {
 			console.error('[updateUser] Update user failed', { userId, user }, err);
 			return;
 		});
+}
+
+export const getUserEmails = async (email: string): Promise<Set<string>> => {  
+	const emails = new Set<string>();
+	try {
+		const users: DbUser[] | undefined = await getUserByAlias(email);
+		if (users) {
+			for (const dbUser of users) {
+				if (dbUser.aliases) {
+					for (const email of dbUser.aliases) {
+						emails.add(email);
+					}
+				}
+				if (dbUser.auth_info) {
+					for (const provider in dbUser.auth_info) {
```

Comment:
unable to use "of" here because AuthInfo is not string or array. Letting this remain as is, removed other "for .. in" and replaced them with "of"

Reply:
Sure, this works. If you actually do not need the provider, you can use `for (cost AuthInfo of Object.values(dbUser.auth_info))` but that would be an overkill. This is fine.